### PR TITLE
Nightly integration tests

### PR DESCRIPTION
**Pull request by anton-l**
In an effort to speed up our `slow`/`push` tests, this PR:
* adds a `@nightly` decorator for tests that can be enabled with `RUN_NIGHTLY=1` 
* suggests `num_inference_steps=3` for all `@slow` tests
* moves end-to-end  image reference-based `@slow` tests to `@nightly`
* suggests `num_inference_steps~50` for all `@nightly` tests

Nightly tests will be run daily at midnight by our CI.

### CODE REVIEW

File: tests/pipelines/stable_diffusion/test_stable_diffusion_img2img.py [hunk 1]
```diff
@@ -212,211 +213,215 @@ def test_stable_diffusion_img2img_num_images_per_prompt(self):
 
 @slow
 @require_torch_gpu
-class StableDiffusionImg2ImgPipelineIntegrationTests(unittest.TestCase):
+class StableDiffusionImg2ImgPipelineSlowTests(unittest.TestCase):
```
Comment (anton-l) [inline tests/pipelines/stable_diffusion/test_stable_diffusion_img2img.py:216]: Img2Img is the most flaky, the results don't match even with 3 steps. If I won't find the cause today, will raise the tolerances

File: .github/workflows/nightly_tests.yml [hunk 2]
```diff
@@ -1,20 +1,119 @@
-name: Nightly integration tests
+name: Nightly tests on main
 
 on:
   schedule:
     - cron: "0 0 * * *" # every day at midnight
```
Comment (patrickvonplaten) [inline .github/workflows/nightly_tests.yml:5]: sounds good!

File: .github/workflows/nightly_tests.yml [hunk 3]
```diff
@@ -1,20 +1,119 @@
...
+#    branches:
+#      - main
 
 env:
   DIFFUSERS_IS_CI: yes
   HF_HOME: /mnt/cache
   OMP_NUM_THREADS: 8
   MKL_NUM_THREADS: 8
-  PYTEST_TIMEOUT: 1000
+  PYTEST_TIMEOUT: 600
```
Comment (patrickvonplaten) [inline .github/workflows/nightly_tests.yml:15]: 10min time out makes sense to me

File: tests/pipelines/stable_diffusion/test_stable_diffusion.py [hunk 4]
```diff
@@ -435,127 +437,137 @@ def test_stable_diffusion_height_width_opt(self):
...
+        return inputs
+
+    def test_stable_diffusion_1_1_pndm(self):
         sd_pipe = StableDiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-1")
         sd_pipe = sd_pipe.to(torch_device)
         sd_pipe.set_progress_bar_config(disable=None)
 
-        prompt = "A painting of a squirrel eating a burger"
-        generator = torch.Generator(device=torch_device).manual_seed(0)
-        with torch.autocast("cuda"):
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion/test_stable_diffusion.py:453]: thanks for cleaning this up

File: tests/pipelines/stable_diffusion/test_stable_diffusion.py [hunk 5]
```diff
@@ -696,17 +664,129 @@ def test_stable_diffusion_pipeline_with_sequential_cpu_offloading(self):
...
+        _ = pipe(**inputs)
 
         mem_bytes = torch.cuda.max_memory_allocated()
         # make sure that less than 2.8 GB is allocated
         assert mem_bytes < 2.8 * 10**9
+
+
+@nightly
+@require_torch_gpu
+class StableDiffusion2PipelineNightlyTests(unittest.TestCase):
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion/test_stable_diffusion.py:685]: nice! But it's SD 1 no? 
```suggestion
class StableDiffusionPipelineNightlyTests(unittest.TestCase):
```

File: tests/pipelines/stable_diffusion/test_stable_diffusion_img2img.py (hunk 1 above)
Comment (patrickvonplaten) [reply tests/pipelines/stable_diffusion/test_stable_diffusion_img2img.py:216]: Sounds good 1e-3 is totally fine. FYI in transformers we only have 5e-3 for one-forward pass tests

**Review by patrickvonplaten — APPROVED** [review]
Super cool! Looks all good to me :-) 

Also definitely enough to start with stable diffusion for now and then maybe open some issue for things that should changed after :-)

File: src/diffusers/utils/testing_utils.py [hunk 6]
```diff
@@ -111,6 +112,16 @@ def slow(test_case):
     return unittest.skipUnless(_run_slow_tests, "test is slow")(test_case)
 
 
+def nightly(test_case):
+    """
+    Decorator marking a test that runs nightly in the diffusers CI.
+
+    Slow tests are skipped by default. Set the RUN_SLOW environment variable to a truthy value to run them.
```
Comment (pcuenca) [inline src/diffusers/utils/testing_utils.py:119]:
```suggestion
    Slow tests are skipped by default. Set the RUN_NIGHTLY environment variable to a truthy value to run them.
```

**Review by pcuenca — APPROVED** [review]
Looking great! Just fixed a typo in the comment for `RUN_NIGHTLY`

**Pull request merged**